### DISCUSSION TITLE
Special optimization for 64 bit unsigned multiplication in Cranelift backend (and possibly LLVM)

### DISCUSSION THREAD

**Discussion by shamatar** [wasmerio/wasmer#2625] [Ideas]
### Summary

I'm interested in pushing to the limits a performance of the code that extensively uses 64x64 -> 128 bit unsigned integer multiplication, and would like to try to implement optimization passes for cases of x86-64 and AOT or JIT compilers available in Wasmer. Hope to get an initial direction suggestions here. 

### Additional details

I'm not a compiler professional or anywhere close to the area, so a description may be not the most precise, but for my case such an optimization should be very powerful and I would like to take a risk to implement it.

**Comment by Hywan** [comment]
Hello,

Thank you for the proposal. Feel free to open a PR! If you need any assistance, we are available.
